### DISCUSSION TITLE
HEIC depth map being output instead of image data

### DISCUSSION THREAD

**Discussion by JonasEklundh** [ImageMagick/ImageMagick#3630] [Help]
So when I run the command `convert photo.heic photo.jpg` I and up with photo-0.jpg and photo-1.jpg where photo-0.jpg is the depth map from the iPhone portrait function. Great, but I don't want that, I want to display the actual photo on my web page, not the depth map. How do I discard the depth map and only output the photo? I've tried with flatten but that didn't work

**Comment by snibgo** [comment]
What version of IM, on what platform?

Try:
```
convert photo.heic[1] photo.jpg
```

**Reply by JonasEklundh** [reply]
Version: ImageMagick 7.0.11-9 Q16 x86_64 2021-04-24  on Mac OS
My production server is on Linux, and I don't see the problem there, so maybe I shouldn't care about it too much.

I can't use your "fix" since the script that presents images won't know beforehand whether or not it's a HEIC file (in fact, the file I'm testing now has a jpeg extension)

**Reply by dlemstra** [reply]
Can you share your image? It is strange that the depth image is the first image. It should be the second image instead.

**Reply by njdoyle** [reply]
Here's an example HEIC with depth map just in case it helps. On my system, the primary image (photo) is the first image and the depth map is the second image, though.

[sloth.heic.zip](https://github.com/ImageMagick/ImageMagick/files/6441769/sloth.heic.zip)

**Comment by dlemstra** [comment]
Thanks for the test image @njdoyle.  I the next version of ImageMagick 7 the depth image will no longer be read by default. It will only be read when `-define heic:depth-image=true` is added before reading the file.

**Comment by JonasEklundh** [comment]
Here is one of the images I experienced this problem with. As far as I can remember, I shot this on my iPhone, edited it a bit and then just uploaded it to my homepage. I may have uploaded it from Photos on the Mac via Safari, and there is maybe something funny going on in that process, making it having a jpeg extension. I know for sure that my upload function wouldn't allow a heic extension. 
[IMG_1984.jpeg.zip](https://github.com/ImageMagick/ImageMagick/files/6445789/IMG_1984.jpeg.zip)
